devel/libjob_queue/src/job_queue.c

```diff
@@ -1249,15 +1244,38 @@ void job_queue_set_pause_off( job_queue_type * job_queue) {
 }
 
 /*
-  An external thread sets the user_exit flag to true, then subsequently the
-  thread managing the queue will see this, and close down the queue.
+  An external thread sets the user_exit flag to true, then
+  subsequently the thread managing the queue will see this, and close
+  down the queue. Will check that the queue is actually running before
+  setting the user_exit flag. If the queue does not change to running
+  state within a timeout limit the user_exit flag is not set, and the
+  function return false.
 */
 
-void job_queue_user_exit( job_queue_type * queue) {
-  queue->user_exit = true;
-}
+bool job_queue_start_user_exit( job_queue_type * queue) {
+  if (!queue->user_exit) {
+    int timeout_limit = 10 * 1000000;  // 10 seconds
+    int usleep_time   =       100000;  // 0.1 second
+    int total_sleep   = 0;
+
+    while (true) {
+      if (queue->running) {
+        queue->user_exit = true;
+        break;
+    }
+      usleep( usleep_time );
+      total_sleep += usleep_time;
 
+      if (total_sleep > timeout_limit)
```

Comment:
> Would it not be more natural to have:

In my opinion no - when there are multiple `break` statements in a loop I like to treat them equally; if the loop was implemented as:

```C
while (total_sleep <= timeout_limit ) { 
    if (queue->running) {
         break;
   }
   ....
}
``` 

The timeout limit and the `queue->running` tests would be treated differently.